src/test/scala/nl/knaw/dans/api/sword2/ResolveFetchItemsSpec.scala

```diff
@@ -94,24 +99,43 @@ class ResolveFetchItemsSpec extends Sword2Fixture with BagStoreFixture {
 
   it should "result in a Failure with a syntactically invalid url in the fetch.txt"  in {
     copyToTargetBagDir(INVALID_URL_BAG)
-    DepositHandler.checkFetchItemUrls(targetBagDir, urlPattern) shouldBe a[Failure[_]]
+    val fetchItemsCheck = DepositHandler.checkFetchItemUrls(targetBagDir, urlPattern)
+    (the [InvalidDepositException] thrownBy fetchItemsCheck.get).getMessage should include("error found in fetch.txt URLs")
   }
 
   it should "result in a Failure with a not allowed url in the fetch.txt"  in {
     copyToTargetBagDir(NOT_ALLOWED_URL_BAG)
-    DepositHandler.checkFetchItemUrls(targetBagDir, urlPattern) shouldBe a[Failure[_]]
+    val fetchItemsCheck = DepositHandler.checkFetchItemUrls(targetBagDir, urlPattern)
+    (the [InvalidDepositException] thrownBy fetchItemsCheck.get).getMessage should include("error found in fetch.txt URLs")
   }
 
   it should "result in a Failure with an empty bag"  in {
     copyToTargetBagDir(NO_DATA_BAG)
     DepositHandler.checkFetchItemUrls(targetBagDir, urlPattern) shouldBe a[Success[_]]
-    DepositHandler.checkBagVirtualValidity(targetBagDir) shouldBe a[Failure[_]]
+    val validity = DepositHandler.checkBagVirtualValidity(targetBagDir)
+    (the [InvalidDepositException] thrownBy validity.get).getMessage should include("Bag does not have any payload manifests")
   }
 
   it should "result in a Failure when the bag-store base-dir doesn't exist"  in {
+    val bagStoreSettings = BagStoreSettings("non/existent/dir", baseUrl)
     copyToTargetBagDir(SIMPLE_SEQUENCE_A)
-    implicit val baseDir = new File("non/existent/dir")
-    DepositHandler.checkBagStoreBaseDir() shouldBe a[Failure[_]]
+    val bagStoreCheck =  DepositHandler.checkBagStoreBaseDir(bagStoreSettings)
+    (the [IOException] thrownBy bagStoreCheck.get).getMessage should include("Bag store base directory")
+  }
+
+  it should "result in a Success when both bag-store base-dir and base-uri are not given, and there are no fetch.txt references to the bag-store"  in {
+    implicit val bagStoreSettings = Option.empty[BagStoreSettings]
+    copyToTargetBagDir(SIMPLE_SEQUENCE_A)
+    DepositHandler.checkFetchItemUrls(targetBagDir, urlPattern) shouldBe a[Success[_]]
+    DepositHandler.checkBagVirtualValidity(targetBagDir) shouldBe a[Success[_]]
+  }
+
+  it should "result in a Failure when both bag-store base-dir and base-uri are not given, and there are fetch.txt references to the bag-store"  in {
```

Comment:
I guess this is the test we talked about, that tries to download files from URL's that are no dereferenceable, that take a long time to fail, right?